brave/brave-core
# Issue 2520: Proxy requests for default google components through componentupdater.brave.com

fix https://github.com/brave/brave-browser/issues/2520

In 

https://github.com/brave/brave-core/blob/1a388e4eb359536813903d834e53611b739be826/chromium_src/components/component_updater/configurator_impl.cc#L9

we added a proxy to retract any user identifiable information for default google components. The change in `kUpdaterDefaultURL` was not propagated to `IsUpdaterURL` that handles update requests for Brave's Components. Updated the code to handle update requests for google components more cleanly.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

1. Run brave
2. Navigate to `brave://components` and verify that CRLSet and Ads components are updated.
3. Browser <data-dir>/CertificateRevocation/ to verify that the CRLSet component is loaded correctly.
4. Navigate to `https://cnn.com` to check if the ads and trackers are being blocked as expected.

## Reviewer Checklist:
